dotnet-authserver/src/TeacherIdentity.AuthServer/Services/DqtEvidence/DqtEvidenceStorageService.cs

```diff
@@ -53,4 +62,25 @@ private async Task<BlobContainerClient> GetBlobContainerClient()
         await blobContainerClient.CreateIfNotExistsAsync();
         return blobContainerClient;
     }
+
+    private async Task<string?> PollForMalwareScanResult(BlobClient blobClient)
+    {
+        await Task.Delay(InitialPollingDelayMs);
+
+        string? malwareScanResult;
+        var malwareScanComplete = false;
+
+        do
+        {
+            await Task.Delay(PollingPeriodMs);
+
+            var blobTags = await blobClient.GetTagsAsync();
+            if (blobTags.Value.Tags.TryGetValue(MicrosoftDefenderMalwareScanKey, out malwareScanResult))
+            {
+                malwareScanComplete = true;
+            }
+        } while (!malwareScanComplete);
```

Comment:
Worth throwing a timeout in here just to stop it spinning forever if something goes wrong?

Reply:
Do we want to handle the exception or just show the default error page?

Reply:
Default error page is fine.